File: file-templates/templates/HelloWorld/View.xctemplate/UIViewObjective-C/___FILEBASENAME___View.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"
#import <Masonry/Masonry.h>

@interface ___VARIABLE_productName:identifier___View ()
@property (nonatomic, strong) UIView *contentView;

@end

@implementation ___VARIABLE_productName___View

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self onInitUIComponents];
    }
    return self;
}

#pragma mark - private

- (void)onInitUIComponents {
    // TODO: implement
    self.contentView.backgroundColor = [UIColor systemCyanColor];
}

- (void)onUpdateUIComponents {
    // TODO: implement
}

- (void)onDataBindings {
    [self onUpdateUIComponents];
    // binding implement
}

#pragma mark - getter/setter

- (UIView *)contentView {
    if (!_contentView) {
        UIView *view = [[UIView alloc] initWithFrame:CGRectZero];
        [self addSubview:view];
        [view mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.mas_equalTo(UIEdgeInsetsMake(0, 0, 0, 0));
        }];
        _contentView = view;
    }
    return _contentView;
}

@end

File: file-templates/templates/MVVM/View.xctemplate/UIViewSwift/___FILEBASENAME___View.swift
//  ___FILEHEADER___
//  Summary: ___VARIABLE_Summary___

import UIKit
import SnapKit

final class ___VARIABLE_productName:identifier___View: UIView {
    
    // MARK: - Private properties
    
    private var viewModel: ___VARIABLE_productName:identifier___ViewModel {
        didSet {
            onDataBindings()
        }
    }
    
    private lazy var contentView: UIView = {
        let view = UIView()
        view.backgroundColor = .clear
        return view
    }()

    // MARK: - Init

    init(viewModel: ___VARIABLE_productName:identifier___ViewModel) {
        self.viewModel = viewModel
        super.init(frame: .zero)
    }

    @available(*, unavailable)
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: - UIComponents

    // 视图加载
    func onInitUIComponents() {
        addSubview(contentView)
    }
    
    // 布局
    func onLayoutUIComponents() {
        contentView.snp.makeConstraints { make in
            make.edges.equalTo(UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
        }
    }
 // UI更新
    func onUpdateUIComponents() {

    }

    // 数据绑定
    func onDataBindings() {
        onUpdateUIComponents()
    }
}